VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/ImplementEqualsAndGetHashCode/ImplementEqualsAndGetHashCodeAnalyzer.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using VSDiagnostics.Utilities;
+
+namespace VSDiagnostics.Diagnostics.General.ImplementEqualsAndGetHashCode
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ImplementEqualsAndGetHashCodeAnalyzer : DiagnosticAnalyzer
+    {
+        private const DiagnosticSeverity Severity = DiagnosticSeverity.Hidden;
+
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.ImplementEqualsAndGetHashCodeAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.ImplementEqualsAndGetHashCodeAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule =>
+            new DiagnosticDescriptor(DiagnosticId.ImplementEqualsAndGetHashCode, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) =>
+                context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration);
+
+        private void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+        {
+            var objectSymbol = context.SemanticModel.Compilation.GetSpecialType(SpecialType.System_Object);
+            IMethodSymbol objectEquals = null;
+            IMethodSymbol objectGetHashCode = null;
+
+            foreach (var symbol in objectSymbol.GetMembers())
+            {
+                if (!(symbol is IMethodSymbol))
+                {
+                    continue;
+                }
+
+                var method = (IMethodSymbol)symbol;
+                if (method.MetadataName == nameof(Equals) && method.Parameters.Count() == 1)
+                {
+                    objectEquals = method;
+                }
+
+                if (method.MetadataName == nameof(GetHashCode) && !method.Parameters.Any())
+                {
+                    objectGetHashCode = method;
+                }
+            }
+
+            if (context.Node is ClassDeclarationSyntax)
+            {
+                var classDeclaration = (ClassDeclarationSyntax) context.Node;
+
+                if (MembersDoNotContainOverridenEqualsAndGetHashCode(context.SemanticModel, classDeclaration.Members, objectEquals, objectGetHashCode) &&
```

Comment:
Ah bummer. I thought I saw an overload that took multiple arguments -- nevermind then.